src/main/java/echoserver/EchoServer.java

```diff
@@ -0,0 +1,54 @@
+package echoserver;
+
+import java.io.PrintWriter;
+import java.io.IOException;
+import java.net.Socket;
+import java.util.Scanner;
+
+public class EchoServer implements Runnable {
+    private final String QUIT = "quit";
```

Comment:
Since this string is constant, we can make this property `static` so that only one copy is created and shared across all instances.

```java
private static final String QUIT = "quit';
```